include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -39,23 +39,53 @@ namespace __ranges
 // walk_n
 //------------------------------------------------------------------------
 
-template <int _N = 0, typename _ExecutionPolicy, typename _Function, typename... _Ranges>
+template <typename _ExecutionPolicy, typename _Function, typename... _Ranges>
 oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy, void>
 __pattern_walk_n(_ExecutionPolicy&& __exec, _Function __f, _Ranges&&... __rngs)
 {
     auto __n = oneapi::dpl::__ranges::__get_first_range_size(__rngs...);
     if (__n > 0)
     {
-        using __new_name = oneapi::dpl::__par_backend_hetero::__new_kernel_name<_ExecutionPolicy, _N>;
-        auto __new_exec =
-            oneapi::dpl::execution::make_hetero_policy<__new_name>(::std::forward<_ExecutionPolicy>(__exec));
-        oneapi::dpl::__par_backend_hetero::__parallel_for(__new_exec,
-                                                          unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f}, __n,
-                                                          ::std::forward<_Ranges>(__rngs)...)
+        oneapi::dpl::__par_backend_hetero::__parallel_for(
+            __exec, unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f}, __n, ::std::forward<_Ranges>(__rngs)...)
             .wait();
     }
 }
 
+//------------------------------------------------------------------------
+// swap
+//------------------------------------------------------------------------
+
+template <typename _Name>
+class __swap1_wrapper
+{
+};
+
+template <typename _Name>
+class __swap2_wrapper
+{
+};
```

Comment:
This looks rather ugly even for two "wrappers", not even speaking about six.

As an idea (not critical, can be made later), I'd consider these classes to be non-templates. Instead, there might be a universal wrapper template (with two parameters) placed nearby the `make_wrapped_policy` function (which should better be called `make_renamed_policy` BTW) and recognized by `_HasDefaultName`. Seems doable at glance, though we'd need to make sure it works well for the "wrapper over a wrapper over a wrapper" cases.

Reply:
I've created an issue not to forget about the suggestion: #530